### PR TITLE
[flashlight-cpu] Fix installation

### LINKED ISSUE #18064: [flashlight-cpu] build failure
**Host Environment**
- OS: Ubuntu 20.04.2 LTS
- Compiler: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

**To Reproduce**
`./vcpkg install flashlight-cpu`

**Failure logs** 
Computing installation plan...
The following packages will be built and installed:
flashlight-cpu[core,fl]:x64-linux -> 0.3#1
Detecting compiler hash for triplet x64-linux...
Could not locate cached archive: /home/skrrohit/.cache/vcpkg/archives/7a/7a2dc37ebb8421a968bf31f3f3e288f81b4352c1.zip
Starting package 1/1: flashlight-cpu:x64-linux
Building package flashlight-cpu[core,fl]:x64-linux...
-- Using cached /home/skrrohit/my_repos/projects/vcpkg/downloads/flashlight-flashlight-626914e79073c5547513de649af706f7e2b796ad.tar.gz
-- Cleaning sources at /home/skrrohit/my_repos/projects/vcpkg/buildtrees/flashlight-cpu/src/f7e2b796ad-c1bedb72d5.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/skrrohit/my_repos/projects/vcpkg/downloads/flashlight-flashlight-626914e79073c5547513de649af706f7e2b796ad.tar.gz
-- Using source at /home/skrrohit/my_repos/projects/vcpkg/buildtrees/flashlight-cpu/src/f7e2b796ad-c1bedb72d5.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:95 (message):

'/home/skrrohit/my_repos/projects/vcpkg/packages/flashlight-cpu_x64-linux/debug/share/flashlight-cpu'
does not exist.
Call Stack (most recent call first):
ports/flashlight-cpu/portfile.cmake:47 (vcpkg_fixup_cmake_targets)
scripts/ports.cmake:141 (include)

Error: Building package flashlight-cpu:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with ./vcpkg update, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
Package: flashlight-cpu:x64-linux
Vcpkg version: 2021-05-05-unknownhash

**Additional context**
Above error is coming because compiler is looking for cmake files in the folder `/home/skrrohit/my_repos/projects/vcpkg/packages/flashlight-cpu_x64-linux/debug/share/flashlight-cpu` but when I checked the location cmake files were present in `home/skrrohit/my_repos/projects/vcpkg/packages/flashlight-cpu_x64-linux/share/flashlight-cpu` so I copied these files to `home/skrrohit/my_repos/projects/vcpkg/packages/flashlight-cpu_x64-linux/debug/share/flashlight-cpu` and flashlight-cpu built successfully. But I  am geeting CMake Error when I am trying to use `flashlight` in my own code. The CMake Error is below: 

```
CMake Error at /home/skrrohit/my_repos/projects/vcpkg/scripts/buildsystems/vcpkg.cmake:664 (_add_executable):
  Target "arr_exe" links to target "flashlight::flashlight" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:8 (add_executable)
```
The contents of the corresponding CMakeLists.txt file is:

```
cmake_minimum_required(VERSION 3.10)
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

find_package(flashlight CONFIG REQUIRED)

add_executable(arr_exe main.cpp)

target_link_libraries(arr_exe PRIVATE flashlight::flashlight)
```

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix installation bug:
```
CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:95 (message):

'/home/skrrohit/my_repos/projects/vcpkg/packages/flashlight-cpu_x64-linux/debug/share/flashlight-cpu'
does not exist.
Call Stack (most recent call first):
```

Fixes #18064.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@SKRohit  Can you please test this PR?

Thanks.

**Comment by JackBoosY** [comment]
Needs to fix the usage also.

**Comment by JackBoosY** [comment]
Should be good, @SKRohit can you please test my changes?

File: versions/f-/flashlight-cpu.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "9a558cf97b737e75e612869995b498709b5a330e",
```
Comment (JackBoosY) [inline versions/f-/flashlight-cpu.json:4]:
```suggestion
      "git-tree": "b37aa1e8b06ce24319b6969ed0fa87664e4ec308",
```

**Review by NancyLi1013 — APPROVED** [review]

**Comment by SKRohit** [comment]
> Should be good, @SKRohit can you please test my changes?

I tried this PR @JackBoosY . I tried to build everything again but got below error:


Computing installation plan...
The following packages will be built and installed:
    flashlight-cpu[core,fl]:x64-linux -> 0.3#2
  * gloo[core,mpi]:x64-linux -> 20201203
  * onednn[core]:x64-linux -> 2.0#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Could not locate cached archive: /root/.cache/vcpkg/archives/47/470c1ac7cc60afc7d5492361c447c1612c6c9000.zip
Could not locate cached archive: /root/.cache/vcpkg/archives/ce/ce5c5b81c07f55b0984433082ad0e1c0be6a4bc3.zip
Could not locate cached archive: /root/.cache/vcpkg/archives/16/163cb435f2fb661cf97c6212e5cc734f68bd51a2.zip
Starting package 1/3: gloo:x64-linux
Building package gloo[core,mpi]:x64-linux...
-- Downloading https://github.com/facebookincubator/gloo/archive/1da21174054eaabbbd189b7f657ea24842d821e2.tar.gz -> facebookincubator-gloo-1da21174054eaabbbd189b7f657ea24842d821e2.tar.gz...
-- Extracting source /home/skrrohit/my_repos/projects/vcpkg/downloads/facebookincubator-gloo-1da21174054eaabbbd189b7f657ea24842d821e2.tar.gz
-- Using source at /home/skrrohit/my_repos/projects/vcpkg/buildtrees/gloo/src/4842d821e2-089f9c85b8.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Installing: /home/skrrohit/my_repos/projects/vcpkg/packages/gloo_x64-linux/share/gloo/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: /root/.cache/vcpkg/archives/47/470c1ac7cc60afc7d5492361c447c1612c6c9000.zip
Building package gloo[core,mpi]:x64-linux... done
Installing package gloo[core,mpi]:x64-linux...
Installing package gloo[core,mpi]:x64-linux... done
Elapsed time for package gloo:x64-linux: 12.76 s
Starting package 2/3: onednn:x64-linux
Building package onednn[core]:x64-linux...
-- Downloading https://github.com/oneapi-src/oneDNN/archive/v2.0.tar.gz -> oneapi-src-oneDNN-v2.0.tar.gz...
-- Extracting source /home/skrrohit/my_repos/projects/vcpkg/downloads/oneapi-src-oneDNN-v2.0.tar.gz
-- Using source at /home/skrrohit/my_repos/projects/vcpkg/buildtrees/onednn/src/v2.0-80e0c80e79.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: /home/skrrohit/my_repos/projects/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /home/skrrohit/my_repos/projects/vcpkg/buildtrees/onednn/x64-linux-dbg
    See logs for more information:
      /home/skrrohit/my_repos/projects/vcpkg/buildtrees/onednn/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:104 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:44 (vcpkg_build_cmake)
  ports/onednn/portfile.cmake:20 (vcpkg_install_cmake)
  scripts/ports.cmake:141 (include)


Error: Building package onednn:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: onednn:x64-linux
  Vcpkg version: 2021-05-05-unknownhash

Additionally, attach any relevant sections from the log files above.

**Comment by JackBoosY** [comment]
@SKRohit Can you please provide _/home/skrrohit/my_repos/projects/vcpkg/buildtrees/onednn/install-x64-linux-dbg-out.log_ ?

**Comment by SKRohit** [comment]
Here it is @JackBoosY . I have uploaded the file in my drive and sharing that link since I was not able to comment the complete output. https://drive.google.com/file/d/1y62eqL2tomzLIg9gwx92aGXXRMZD7-4i/view?usp=sharing

**Comment by JackBoosY** [comment]
@SKRohit 
```
collect2: fatal error: ld terminated with signal 9 [Killed]
```
I think you don't have enough memory.

**Comment by SKRohit** [comment]
@JackBoosY I have around 112GB left on my SSD and my RAM is 16GB. Btw after trying the command `./vcpkg install flashlight-cpu' I am getting this error. How could I start building `onednn` again?

Computing installation plan...
The following packages will be built and installed:
    flashlight-cpu[core,fl]:x64-linux -> 0.3#1
  * onednn[core]:x64-linux -> 2.0#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Error: while detecting compiler information:
The log content at /home/skrrohit/my_repos/projects/vcpkg/buildtrees/detect_compiler/stdout-x64-linux.log is:
-- Configuring x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: /home/skrrohit/my_repos/projects/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/bin/cmake /home/skrrohit/my_repos/projects/vcpkg/scripts/detect_compiler -DCMAKE_MAKE_PROGRAM=/home/skrrohit/my_repos/projects/vcpkg/downloads/tools/ninja-1.10.1-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/skrrohit/my_repos/projects/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/skrrohit/my_repos/projects/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/skrrohit/my_repos/projects/vcpkg -D_VCPKG_INSTALLED_DIR=/home/skrrohit/my_repos/projects/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/skrrohit/my_repos/projects/vcpkg/packages/detect_compiler_x64-linux
    Working Directory: /home/skrrohit/my_repos/projects/vcpkg/buildtrees/detect_compiler/x64-linux-rel
    Error code: 1
    See logs for more information:
      /home/skrrohit/my_repos/projects/vcpkg/buildtrees/detect_compiler/config-x64-linux-rel-out.log
      /home/skrrohit/my_repos/projects/vcpkg/buildtrees/detect_compiler/config-x64-linux-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:342 (vcpkg_execute_required_process)
  scripts/detect_compiler/portfile.cmake:18 (vcpkg_configure_cmake)
  scripts/ports.cmake:141 (include)

**Comment by JackBoosY** [comment]
@SKRohit Can you please provide  _/home/skrrohit/my_repos/projects/vcpkg/buildtrees/detect_compiler/config-x64-linux-rel-out.log_ and _/home/skrrohit/my_repos/projects/vcpkg/buildtrees/detect_compiler/config-x64-linux-rel-err.log_?

**Comment by JackBoosY** [comment]
Ping @SKRohit for response.

**Comment by strega-nil-ms** [comment]
Thanks @JackBoosY :)

**Pull request merged**